bc_obps/registration/tests/endpoints/v2/test_operators_endpoint.py

```diff
@@ -0,0 +1,104 @@
+    endpoint = CommonTestSetup.base_endpoint + "v2/operators"
+
+    # AUTHORIZATION
+
+    def test_unauthorized_roles_cannot_list_operators_v2(self):
+        response = TestUtils.mock_get_with_auth_role(self, 'cas_pending', custom_reverse_lazy("list_operators_v2"))
+        assert response.status_code == 401
+
+    def test_operators_endpoint_list_operators_v2_paginated(self):
+        # Create 60 operators with unique corp numbers
+        for i in range(60):
+            operator_baker()
+        # Get the default page 1 response
+        response = TestUtils.mock_get_with_auth_role(self, "cas_admin")
+        assert response.status_code == 200
+        response_data = response.json().get('data')
+        # save the id of the first paginated response item
+        page_1_response_id = response_data[0].get('id')
+        assert len(response_data) == PAGE_SIZE
+        # Get the page 2 response
+        response = TestUtils.mock_get_with_auth_role(
+            self,
+            "cas_admin",
+            custom_reverse_lazy('list_operators_v2') + "?page=2&sort_field=created_at&sort_order=desc",
+        )
+        assert response.status_code == 200
+        response_data = response.json().get('data')
+        # save the id of the first paginated response item
+        page_2_response_id = response_data[0].get('id')
+        assert len(response_data) == PAGE_SIZE
+        # assert that the first item in the page 1 response is not the same as the first item in the page 2 response
+        assert page_1_response_id != page_2_response_id
+
+        # Get the page 2 response but with a different sort order
+        response = TestUtils.mock_get_with_auth_role(
+            self,
+            "cas_admin",
+            custom_reverse_lazy('list_operators_v2') + "?page=2&sort_field=created_at&sort_order=asc",
+        )
+        assert response.status_code == 200
+        response_data = response.json().get('data')
+        # save the id of the first paginated response item
+        page_2_response_id_reverse = response_data[0].get('id')
+        assert len(response_data) == PAGE_SIZE
+        # assert that the first item in the page 2 response is not the same as the first item in the page 2 response with reversed order
+        assert page_2_response_id != page_2_response_id_reverse
+
+    def test_operators_endpoint_list_operators_v2_with_filter(self):
+        operator_baker({"legal_name": 'gouda'})
+        operator_baker({"cra_business_number": '118675309'})
+        for i in range(60):
+            operator_baker()
+
+        # Get the default page 1 response
+        response = TestUtils.mock_get_with_auth_role(
+            self, "cas_admin", custom_reverse_lazy('list_operators_v2') + "?legal_name=gouda"
+        )
+        assert response.status_code == 200
+        response_data = response.json().get('data')
+        assert len(response_data) == 1
+        for item in response_data:
+            assert item.get('legal_name') == 'gouda'
```

Comment:
We can use an index instead of a for-loop since we only have one item with that name.
```python
response_data[0].get('legal_name') == 'gouda'
```

Reply:
@gdalcengio what a very un-Italian choice of cheese. Tsk tsk.

Reply:
@andrea-williams I took it from operations endpoint and it is a very gouda cheese, but you're right. Will change it.